packages/ui-kit/src/design-system.ts

```diff
@@ -7,13 +7,76 @@ const asideColumnWidth = pageWidthSmall;
 const navbarWidth = '224px';
 const navbarWidthSmall = '200px';
 
-export const pxToRem = (px, suffix = 'rem') => {
+export const pxToRem = (px: string, suffix = 'rem') => {
   const pxNumber = px.replace(/([0-9]+)px$/, '$1');
   const remNumber = parseInt(pxNumber, 10) / rootFontSizeNumber;
   return `${remNumber}${suffix}`;
 };
 
-export const colors = {
+// explicit types for the color part because the structure allows multiple instances
+// (light / dark,  primary / secondary for code blocks).
+// The rest of the design system is enough to be inferred
+export type ColorKey = keyof ColorSet;
+export type ColorSet = {
```

Comment:
Usually for constants you should use `as const`. There is no need to have a separate type.

```ts
export const colors = {
  // ...
} as const;
```

Reply:
The types are here because I had issues letting the code that uses it understand that "Primary" and "secondary" code block style are the same identical interface. 

this code in `code-block.tsx`:
```
  const codeBlockTheme = {
    codeBlockColors:
      colors.light.codeBlocks[props.secondaryTheme ? 'secondary' : 'primary'],
  };
```

Reply:
I pushed a commit to use `as const`. https://github.com/commercetools/commercetools-docs-kit/pull/1217/commits/9d7c327010e3bafd09aa914ce85fab3ed925b69d